### ISSUE TITLE
Combined Lifetime Bounds Cause Compiler to Crash Unexpectedly

### ISSUE THREAD

**Issue by dpchamps** [rust-lang/rust#69783]
**Problem**

A Certain combination of lifetime bounds cause compiler to exit with a non-zero status.

Appears to be reproducible only on Linux. Compiles as expected on MacOs with similar versions of rustup and cargo.

**Steps**

```
pub trait SomeTrait {}

struct SomeGeneric<T>{
    field : T
}

pub type SomeType<T: 'static + SomeTrait + ?Sized> = SomeGeneric<T>;
```

1. Attempt to compile above code via `cargo build`

**Notes**

Removing any one of the above bounds results in a successful compilation, e.g:

| type                                                                 | compiles correctly |
|----------------------------------------------------------------------|--------------------|
| pub type SomeType<T: 'static + SomeTrait > = SomeGeneric<T>;         | ✔️                  |
| pub type SomeType<T: 'static + ?Sized> = SomeGeneric<T>;             | ✔️                  |
| pub type SomeType<T: SomeTrait + ?Sized> = SomeGeneric<T>;           | ✔️                  |
| pub type SomeType<T: 'static + SomeTrait + ?Sized> = SomeGeneric<T>; | ❌                  |

___

| cargo     | 1.41.0 (626f0f40e 2019-12-03)   |
|-----------|---------------------------------|
| rustup    | 1.21.1 (7832b2ebe 2019-12-20)   |
| toolchain | stable-x86_64-unknown-linux-gnu |

```
#  lsb_release -a
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.3 LTS
Release:        18.04
Codename:       bionic
```

**Comment by alexcrichton** [comment]
Thanks for the report! I've transferred this issue to rust-lang/rust since I don't think this is an issue with Cargo itself.

**Comment by jonas-schievink** [comment]
Fixed on beta/nightly

**Comment by dpchamps** [comment]
Thanks all!

**Comment by ehuss** [comment]
Fixed PR is #67880.
